src/hotspot/share/opto/loopnode.cpp

```diff
@@ -4695,6 +4678,18 @@ void PhaseIdealLoop::build_and_optimize() {
       }
     }
   }
+
+  // Perform loop vectorization with vector masks
+  if (UseMaskedLoop && Matcher::has_predicated_vectors() &&
+      C->has_loops() && !C->major_progress()) {
+    VectorMaskedLoop vml(this);
+    for (LoopTreeIterator iter(_ltree_root); !iter.done(); iter.next()) {
+      IdealLoopTree* lpt = iter.current();
+      if (lpt->is_counted() && lpt->is_innermost()) {
```

Comment:
Is this applied to all innermost counted loops? Or only post-loops?

Reply:
Ah, you do the check inside. Why not lift it out and assert inside?

Reply:
I have lift the checks out in commit 2 and added assertions inside.